compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java

```diff
@@ -151,25 +162,90 @@
         args.add(methodNameLiteral);
         args.add(resourcePathLiteral);
 
-        BLangInvocation invocationExpr = ASTBuilderUtil
-                .createInvocationExprForMethod(pos, filterInvocationSymbol, args, symResolver);
+        BLangInvocation invocationExpr =
+                ASTBuilderUtil.createInvocationExprForMethod(pos, invocationSymbol, args, symResolver);
+        BLangSimpleVariableDef result = ASTBuilderUtil.createVariableDef(pos,
+                 ASTBuilderUtil.createVariable(pos, "$temp$http$desugar$result",
+                                               symTable.anyType, invocationExpr, null));
+        List<BLangStatement> statements = getFunctionBodyStatementList(functionNode);
+        statements.add(0, result);
+
+        BVarSymbol resultSymbol = new BVarSymbol(0, names.fromIdNode(result.var.name), env.enclPkg.packageID,
+                                                 result.var.type, resourceNode.symbol, pos, VIRTUAL);
+        resourceNode.symbol.scope.define(resultSymbol.name, resultSymbol);
+        result.var.symbol = resultSymbol;
+    }
+
+    void addWebSocketAuthDesugarFunctionInvocation(BLangFunction functionNode, SymbolEnv env) {
```

Comment:
It looks like these 2 methods do the kind of same things, shall we try to extract common code into methods and re-use them in this 2 methods.

Reply:
Agree. Updated into a single method and there will be few additional steps if the package is HTTP. Else, everything will be executed.